src/twisted/internet/defer.py

```diff
@@ -674,14 +784,16 @@ def _runCallbacks(self):
                 except BaseException:
                     # Including full frame information in the Failure is quite
                     # expensive, so we avoid it unless self.debug is set.
-                    current.result = failure.Failure(captureVars=self.debug)
+                    current.result = Failure(captureVars=self.debug)
                 else:
                     if isinstance(current.result, Deferred):
                         # The result is another Deferred.  If it has a result,
                         # we can take it and keep going.
-                        resultResult = getattr(current.result, "result", _NO_RESULT)
+                        resultResult = getattr(
+                            current.result, "result", _Sentinel._NO_RESULT
+                        )
```

Comment:
This code path is extremely hot, so I'd be a little wary of adding the two extra `getattr` calls here. (I wish speed center could quickly just, you know, tell us the performance impact instead of guessing like this, but c'est la vie)

Reply:
(In other words, let's still cache `_NO_RESULT` in a global even if we want its type to be an enum for signature friendliness)

Reply:
Done: https://github.com/twisted/twisted/pull/1448/commits/f35df2083e6a78c13230ca689832654d0d636d9f